heudiconv/utils.py

```diff
@@ -662,32 +664,107 @@ def get_datetime(date: str, time: str, *, microseconds: bool = True) -> str:
         # add dummy microseconds if not available for strptime to parse
         time += ".000000"
     td = time + ":" + date
-    datetime_str = datetime.strptime(td, "%H%M%S.%f:%Y%m%d").isoformat()
+    datetime_str = datetime.datetime.strptime(td, "%H%M%S.%f:%Y%m%d").isoformat()
     if not microseconds:
         datetime_str = datetime_str.split(".", 1)[0]
     return datetime_str
 
+def datetime_utc_offset(datetime_obj: datetime, utc_offset: str):
+    """set the datetime's tzinfo by parsing an utc offset string"""
+    sign, hours, minutes = re.match(r"([+\-]?)(\d{2})(\d{2})", utc_offset).groups()
+    sign = -1 if sign == '-' else 1
+    hours, minutes = int(hours), int(minutes)
+    tzinfo = datetime.timezone(sign * datetime.timedelta(hours=hours, minutes=minutes))
+    return datetime_obj.replace(tzinfo=tzinfo)
 
-def strptime_micr(date_string: str, fmt: str) -> datetime:
```

Comment:
Please keep prior implementation around, but add a deprecation warning. It could be that someone used it in their heuristic, so we better do not just rip it from under them.